### DISCUSSION TITLE
Locked roto capablity / fan to 100%

### DISCUSSION THREAD

**Discussion by Zitt** [tjko/fanpico#107] [Q&A]
@tjko Related to the #87 - Locked rotor commit...
I have a fan pico being planned for use in an APC UPS which has noisy arse fans in it.
I designed a application specific board with takes it's 24volt fans  and converts them to a temperature sensitive 12VDC configuration. I plan to share the full board once I've added the full capability.

The problem I currently have is that I've added some sense electronics to my board to determine if the APC has set the fan's driving voltage to >17VDC. 
In normal operation; on AC power; the UPS seems to drive the fan headers to around 17.5V.
On battery operation; the UPS drives the fans to their full 24VDC operation mode.

What I need my application fanpico to do is temperature sense and quiet the fans while the ups is on AC mode. 
When the battery mode kicks in; the fanpico should drive all fans to 100% until AC is restored.

Could the lock roto capabity be abused to do this?

Here's implementation of the sense circuit:
![fanPico24Vsense](https://github.com/user-attachments/assets/4f94e74d-130b-4597-8004-c97e024c708c)

Where +UB is the 24Vfan rail.
When the fan is below 70%; the GP13,15,17, and 19 are all pulled low.
When the fan is above 70% (or on battery power) those GPIOs are pulled high.

Originally; I was just planning on adding some logic at the appropriate place in the Firmware... but I'm kindof hoping the locked router addition you made can be abused to accomplish the same goal. 
Thoughts?

**Comment by tjko** [comment]
@Zitt, currently the locked rotor support work the opposite way. Reading LRA signal on tachometer pin on FAN ports (or sending LRA signal out on MBFAN tachometer pin).

But you should be able to feed 5V (3.3V probably would work too) high/low signal to PWM pins on MBFAN ports and it should read as 100% PWM with high signal, and 0% PWM on low signal.

I've been thinking of expanding the CONF:FANx:SOURCE command to work bit like the CONF:MBFANx:SOURCE, so it would be possible to configure a FAN get "max" (or "min") PWM signal from multiple sources...   Like currently its possible to have two "CPU" fans and feed the lowest RPM speed from those two fans back to motherboard via one MBFAN port...


So could issue command like:
CONF:FAN1:SOURCE MAX,MBFAN1,SENSOR2

And as long as tachometer pin is low on MBFAN1 connector fans would run based on temperature from SENSOR2, but
as soon as tachometer pin goes high, fans would run at 100%....




Currently there is no support to specify function (MIN/MAX) as "source" for fan control (but support exists on MBFAN port side).

**Comment by Zitt** [comment]
Ok. I need to think about how to best proceed since it sounds like the support isn't there right now. 
Appreciate the feedback.

**Comment by Zitt** [comment]
I made some progress this weekend; but wasn't able to make it work like I expected.

I have GP13, 15, 17, and 19 all tied together - expecting those PWM signals to be an input. 
As you can see in the previous schematics; I have the FAN100% signal pulled high to 3v3 via a 10k external pull up.
Yet; when I measured the signal at ~0.7 even when the comparator has 5.5V on the + and 4.9V on the -.

Are the MBFANx not configured as inputs?

**Comment by tjko** [comment]
Those GPIO pins are configured as inputs that are used to "read" PWM signal from MFANx connectors. So they shouldn't affect the voltage (as GPIO pins should be in high-impedance state).

If would seem like there must be something (stronger than that 10k pull-up to 3.3V) pulling that signal down, if you measure only 0.7V...

**Comment by Zitt** [comment]
yeah; that's what I was thinking too. I'll do some more troubleshooting. 

I'm using some high density / fine pitch 
TMUX 4052BQB 
chips on that single; I wonder if I have a solder bridge under the package.
